googletest/test/gtest_unittest.cc

```diff
@@ -146,6 +146,27 @@ TEST_F(StreamingListenerTest, OnTestPartResult) {
       *output());
 }
 
+TEST_F(StreamingListenerTest, OnTestPartResultWithNonFatalFailure) {
+  *output() = "";
+  streamer_.OnTestPartResult(TestPartResult(
+      TestPartResult::kNonFatalFailure, "foo.cc", 42, "failed=\n&%"));
+
+  // Meta characters in the failure message should be properly escaped.
+  EXPECT_EQ(
+      "event=TestPartResult&file=foo.cc&line=42&message=failed%3D%0A%26%25\n",
+      *output());
+}
+
+TEST_F(StreamingListenerTest, OnTestPartResultWithSkip) {
+  *output() = "";
+  streamer_.OnTestPartResult(TestPartResult(
+      TestPartResult::kSkip, "foo.cc", 42, "failed=\n&%"));
```

Comment:
No

Reply:
What about this code snippet that makes you say "no"?